src/unit_tests/wazuh_modules_scheduling/test_aws_scheduling.c

```diff
@@ -259,36 +263,32 @@ void test_read_scheduling_interval_configuration() {
         "   <aws_profile>default</aws_profile>\n"
         "</bucket>\n"
     ;
-    wmodule *module = calloc(1, sizeof(wmodule));;
-    OS_XML xml;
-    XML_NODE nodes = string_to_xml_node(string, &xml);
-    assert_int_equal(wm_aws_read(&xml, nodes, module), 0);
-    wm_aws *module_data = (wm_aws*)module->data;
+    test_structure *test = *state;
+    test->nodes = string_to_xml_node(string, &(test->xml));
+    assert_int_equal(wm_aws_read(&(test->xml), test->nodes, test->module), 0);
+    wm_aws *module_data = (wm_aws*)test->module->data;
     assert_int_equal(module_data->scan_config.scan_day, 0);
     assert_int_equal(module_data->scan_config.interval, 600);
     assert_int_equal(module_data->scan_config.month_interval, false);
     assert_int_equal(module_data->scan_config.scan_wday, -1);
-    OS_ClearNode(nodes);
-    OS_ClearXML(&xml);
-    wmodule_cleanup(module);
 }
 
 int main(void) {
     const struct CMUnitTest tests_with_startup[] = {
-        cmocka_unit_test(test_interval_execution),
-        cmocka_unit_test(test_day_of_month),
-        cmocka_unit_test(test_day_of_week),
-        cmocka_unit_test(test_time_of_day)
+        cmocka_unit_test_setup_teardown(test_interval_execution, NULL, teardown_test_executions),
+        cmocka_unit_test_setup_teardown(test_day_of_month, NULL, teardown_test_executions),
+        cmocka_unit_test_setup_teardown(test_day_of_week, NULL, teardown_test_executions),
+        cmocka_unit_test_setup_teardown(test_time_of_day, NULL, teardown_test_executions)
```

Comment:
`cmocka_unit_test_setup_teardown` can be replaced by `cmocka_unit_test_teardown` if no setup function is needed